### ISSUE TITLE
gfx upgrade causing panics during window creation for OpenGL 2.1

### ISSUE THREAD

**Issue by SonOfFlyingPig** [PistonDevelopers/piston_window#143]
Trying to build a simple example ([main.rs.txt](https://github.com/PistonDevelopers/piston_window/files/254101/main.rs.txt)) with OpenGL 2.1 causes a panic while creating the window ([error.txt](https://github.com/PistonDevelopers/piston_window/files/254100/error.txt)). This happens on both my Fedora 23 virtualbox VM and on its Windows 10 x64 host.

It works fine for Windows when I do not use 2.1 . The last working version of piston_window was 0.45.0, before the gfx upgrade.

**Comment by Bluebird45** [comment]
Interesting! I could not reproduce any such error, just now.

I created the [simple example](https://github.com/PistonDevelopers/piston "simple example"), with the dependency specified there of piston_window 0.44.0. Ran OK. Then I changed the dependency  to 0.47.0 and did "cargo clean" and "cargo update" (gfx was one of the updates). It still ran OK. So it must be something else than gfx.

This is with OpenGL 2.1, Debian 8.4 (Jessie) 64-bit, Rust 1.8 (Stable), using Rustup.

**Comment by SonOfFlyingPig** [comment]
And you explicitly set it to 2.1 in your code (the one in the piston readme does not)?

**Comment by Bluebird45** [comment]
Yes, I pasted [your code](https://github.com/PistonDevelopers/piston_window/files/254101/main.rs.txt "your code") in place of the one in the readme, although they differ only in setting it to 2.1.

**Comment by jeandudey** [comment]
I have the same panic on Ubuntu 16.04 using OpenGL 2.1 explicitly.

**Comment by Bluebird45** [comment]
@jeandudey Is that a default installation of Ubuntu 16.04, or are there any customizations, for example of desktop environment? What version of Rust? I'll install it on a spare machine.

**Comment by jeandudey** [comment]
Fresh version of Ubuntu (i've only modified the wallpaper for now :smile:) and `rustc 1.8.0 (db2939409 2016-04-11)` installed with multirust.

**Comment by SonOfFlyingPig** [comment]
@Bluebird45 just FYI in my case it was Rust 1.8 on both the Fedora VM and my Win10 machine

**Comment by Bluebird45** [comment]
@jeandudey @SonOfFlyingPig I don't know if this is relevant to the issue, and I still haven't had time to set up a test machine, but could you confirm that it still exists after the [patch](https://github.com/PistonDevelopers/pistondevelopers.github.io/pull/117/commits/91c5785b050f36056034b0df442ba7c0d5861679 "patch) to the example earlier today?

**Comment by jeandudey** [comment]
Testing on the same computer with Windows 10 fails with: `Failed to build PistonWindow: Couldn't find any pixel format that matches the criterias.`

**Comment by pengowen123** [comment]
Fails with the same error for me on Windows 7. It worked fine with 0.44.0, but after upgrading I get the error.

**Comment by C-Bouthoorn** [comment]
@jeandudey That might be because of the issue mesa had with sRGB settings. That issue should be fixed with the latest 18.* version. See https://github.com/PistonDevelopers/piston/issues/1202 for more information.
